src/JsonSchema/SchemaFactory.php

```diff
@@ -230,14 +233,19 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
                 break;
             }
 
-            if ($type->isNullable()) {
-                $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
-            } else {
-                $propertySchema['$ref'] = $subSchema['$ref'];
-            }
+            $refs[] = ['$ref' => $subSchema['$ref']];
+            $isNullable = $isNullable ?? $type->isNullable();
+        }
+
+        if ($isNullable) {
+            $refs[] = ['type' => 'null'];
+        }
 
+        if (\count($refs) > 1) {
+            $propertySchema['anyOf'] = $refs;
             unset($propertySchema['type']);
```

Comment:
If `anyOf` or `$ref` is specified, `type` must be removed.

Reply:
I'm confused again. :innocent: The `unset()` is removing `type` isn't it?

Reply:
Yep!

Reply:
But you can move this line before (or after) the if to prevent duplication with line 249

Reply:
Got it! Just to check your meaning though, as we don't want to unset `type` if `\count($refs) > 0`, is this whasat you're thinking?

```php
        if (\count($refs) > 1) {
            $propertySchema['anyOf'] = $refs;
        } elseif (1 === \count($refs)) {
            $propertySchema['$ref'] = $refs[0]['$ref'];
        }
        if (\count($refs) > 0) {
            unset($propertySchema['type']);
        }
```

For extra context, a resource with a `?int $property;` will have an empty `$refs` array, but `$propertySchema['type']` will contain `['integer', 'null']`

Reply:
Ah nice catch! I didn't think about this specific use case.
In this case, indeed I would prefer to keep duplication to avoid useless code complexity!
You can keep your code as it is